doc/api/errors.markdown

```diff
@@ -2,222 +2,250 @@
 
 <!--type=misc-->
 
-Errors generated by Node.js fall into two categories: JavaScript errors and system
-errors. All errors inherit from or are instances of JavaScript's [`Error`][]
-class and are guaranteed to provide *at least* the attributes available on that
-class.
-
-When an operation is not permitted due to language-syntax or
-language-runtime-level reasons, a **JavaScript error** is generated and thrown
-as an **exception**. If an operation is not allowed due to system-level
-restrictions, a **system error** is generated. Client code is then given the
-opportunity to **intercept** this error based on how the API **propagates** it.
-
-The style of API called determines how generated errors are handed back, or
-**propagated**, to client code, which in turn informs how the client may **intercept**
-the error. Exceptions can be intercepted using the [`try / catch` construct][];
-other propagation strategies are covered [below][].
+Applications running in Node.js will generally experience four categories of
+errors:
+
+- JavaScript errors caused by improper JavaScript language syntax
+  ([`SyntaxError`][]), use of undefined variables ([`ReferenceError`][]),
+  passing arguments of the wrong type ([`TypeError`][]), and so on;
```

Comment:
and so on here is a bit non-descriptive for documentation. If there are other JavaScript errors we should likely list them right?